File: workspace/solutions/kotlin/lib/src/main/kotlin/s2/s1/s0/s4/s_2104_00/s_2104_00.kt
/*
 * @lc app=leetcode.cn id=2104 lang=kotlin
 *
 * [2104] 子数组范围和
 */

package s2.s1.s0.s4.s_2104_00

// @lc code=start
class Solution {
    fun subArrayRanges(nums: IntArray): Long {
        val n = nums.size
        var result = 0L
        for (i in 0..<(n - 1)) {
            var min = nums[i]
            var max = nums[i]
            for (j in (i + 1)..<n) {
                min = Math.min(min, nums[j])
                max = Math.max(max, nums[j])
                result += max - min
            }
        }
        return result
    }
}
// @lc code=end

File: workspace/solutions/ecmascript/src/s2/s1/s0/s4/s_2104_00.js
/*
 * @lc app=leetcode.cn id=2104 lang=javascript
 *
 * [2104] 子数组范围和
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
const subArrayRanges = function (nums) {
    const n = nums.length;
    let result = 0;
    for (let i = 0; i < n - 1; i++) {
        let min = nums[i] ?? 0;
        let max = nums[i] ?? 0;
        for (let j = i + 1; j < n; j++) {
            min = Math.min(min, nums[j] ?? 0);
            max = Math.max(max, nums[j] ?? 0);
            result += max - min;
        }
    }
    return result;
};
// @lc code=end

export default subArrayRanges;
